src/drive/targets/mobile/InitAppMobile.jsx

```diff
@@ -251,6 +263,72 @@ class InitAppMobile {
     return
   }
 
+  migrateOfflineFiles = async () => {
+    try {
+      const alreadyMigrated = await localforage.getItem(
+        'offlineFilesMigration-finished'
+      )
+      if (alreadyMigrated || !isWifi()) return
+
+      const store = await this.getStore()
+      const client = await this.getClient()
+
+      const availableOfflineIds = getAvailableOfflineIds(store.getState())
+      if (!availableOfflineIds || availableOfflineIds.length < 1) {
+        await localforage.setItem('offlineFilesMigration-finished', true)
+        return
+      }
+
+      // stop media backup
+      store.dispatch(cancelMediaBackup())
+
+      let error = false
+
+      for (const fileId of availableOfflineIds) {
+        try {
+          const { data } = await client.query(
+            Q('io.cozy.files').getById(fileId)
+          )
+          const file = data && data.data
+          await saveOfflineFileCopy(file, client)
+          await updateLocalStorageItem(
+            'offlineFilesMigration-migratedFileIds',
+            fileId
+          )
+        } catch (e) {
+          error = true
+
+          store.dispatch(markAsUnavailableOffline(fileId))
+          await updateLocalStorageItem(
+            'offlineFilesMigration-notMigratedFileIds',
+            fileId
+          )
+        }
+      }
+
+      // restart media backup
+      store.dispatch(startMediaBackup())
+
+      if (error) {
+        const migratedFileIds = await localforage.getItem(
+          'offlineFilesMigration-migratedFileIds'
+        )
+        const notMigratedFileIds = await localforage.getItem(
+          'offlineFilesMigration-notMigratedFileIds'
+        )
+
+        throw new Error(
+          `${migratedFileIds.length} of ${notMigratedFileIds.length +
+            migratedFileIds.length} files have been migrated successfully. The files with these ids have not been migrated: ${notMigratedFileIds}`
+        )
+      }
+    } catch (e) {
+      logger.error(`migrateOfflineFiles finished with error : ${e}`)
+    }
+
+    await localforage.setItem('offlineFilesMigration-finished', true)
+  }
```

Comment:
It brings a lot of code to InitApp. Can we move the code to a specific files/folders? 

Having a `migrations` folder with `migrateOfflineFile.jsx` can be a good start. Since we can imagine having something, in the future, that run the needed migrations automatically (like you have with some ORM like Doctrine or alembic)

Reply:
Oh oui avec plaisir !